File: CompleteSolutions/pishgooe_khar.cpp
// https://quera.org/problemset/4066/
// writed with chatGPT python to cpp

#include <iostream>
#include <string>
#include <map>
#include <vector>

using namespace std;

int main() {
    int n, m;
    cin >> n >> m;

    map<string, string> dictionary;
    string key, val;
    for (int i = 0; i < n; i++) {
        cin >> key >> val;
        dictionary[key] = val;
    }

    vector<string> txt(m);
    for (int i = 0; i < m; i++) {
        cin >> txt[i];
        if (dictionary.find(txt[i]) != dictionary.end()) {
            txt[i] = dictionary[txt[i]] + " kachal!";
        } else {
            txt[i] = "kachal!";
        }
    }

    for (auto& t : txt) {
        cout << t << " ";
    }
    cout << endl;

    return 0;
}

File: CompleteSolutions/pishgooe_khar.py
# https://quera.org/problemset/4066/

dictionary = {key : val for key, val in [input().split() for i in range(list(map(int, input().split()))[0])]}
txt = input().split()
print(*[dictionary[txt[i]] + " kachal!" if txt[i] in dictionary.keys() else "kachal!" for i in range(len(txt))], sep=' ')


# n, m = list(map(int, input().split()))
# dictionary = {key : val for key, val in [input().split() for i in range(n)]}
# # print(dictionary)
# txt = input().split()
# # for i in range(len(txt)):
# #     if txt[i] in dictionary.keys():
# #         txt[i] = dictionary[txt[i]] + " kachal!"
# #     else:
# #         txt[i] = "kachal!"
# print(*txt, sep=' ')